src/ICS26Router.sol

```diff
@@ -205,6 +214,24 @@ contract ICS26Router is IICS26Router, IICS26RouterErrors, Ownable, ReentrancyGua
 
         ICS02_CLIENT.getClient(msg_.packet.sourceChannel).membership(membershipMsg);
 
+        // ackPacket will no-op if the packet commitment does not exist
+        try IBC_STORE.deletePacketCommitment(msg_.packet) returns (bytes32 storedCommitment) {
+            if (storedCommitment != ICS24Host.packetCommitmentBytes32(msg_.packet)) {
+                revert IBCPacketCommitmentMismatch(storedCommitment, ICS24Host.packetCommitmentBytes32(msg_.packet));
+            }
+        } catch (bytes memory reason) {
+            if (bytes4(reason) == IICS24HostErrors.IBCPacketCommitmentNotFound.selector) {
+                emit Noop();
+                return; // no-op since the packet commitment already deleted
+            } else {
+                // reverts with the same reason
```

Comment:
I assume the reason for this is that there is no "normal" way to rethrow/revert?

Reply:
Yes, I think there is not. This is also what they do in openzeppelin contracts